### PR TITLE
Analytics: track tx creations and safe views

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

As per [this doc](https://www.notion.so/safe-global/GA-event-relabelling-98aba2185e684bcdab67a295a5b616cf), we want to track key events in a new way, and for that we were missing a `transaction_created` and `safe_opened` events.

So I've instrumented all of our tx flows to emit a new CREATED event + the corresponding tx type from a pre-defined list.

Also, a few changes:
* Added an event when a distinct Safe Account is opened.
* Restored a `Simulate tx` event that we lost in the tx flow redesign
* Removed the unused `Proposed tx`
* Renamed `Execute transaction` to `Toggle execute transaction` because it's actually triggered on the Execute checkbox

### `Safe viewed` event
* Open a Safe, a GTM event should be triggerred
* Switch to another Safe, the event should trigger again
* Browsing pages within the same Safe should not trigger it

### `Transaction created` event
* Create all types of transactions possible via our UI
* Each tx creation  should be tracked along with its type from thsis list

```
  // Owner txs
  owner_add = 'owner_add',
  owner_remove = 'owner_remove',
  owner_swap = 'owner_swap',
  owner_threshold_change = 'owner_threshold_change',

  // Module txs
  guard_remove = 'guard_remove',
  module_remove = 'module_remove',
  spending_limit_remove = 'spending_limit_remove',
  spending_limit_add = 'spending_limit_add',

  // Safe txs
  safe_update = 'safe_update',

  // Transfers
  transfer_token = 'transfer_token',
  transfer_nft = 'transfer_nft',

  // Other
  batch = 'batch',
  rejection = 'rejection',
  typed_message = 'typed_message',
  safeapps = 'safeapps',
  walletconnect = 'walletconnect',
```

### CODE REVIEW

**Review by alicecha — APPROVED** [review]
Looks good to me.

File: src/services/analytics/events/overview.ts [hunk 1]
```diff
@@ -96,4 +96,8 @@ export const OVERVIEW_EVENTS = {
     action: 'Click on SEP5 allocation button',
     category: OVERVIEW_CATEGORY,
   },
+  SAFE_VIEWED: {
+    action: 'Safe viewed',
+    category: OVERVIEW_CATEGORY,
+  },
```
[the remark this answers is not in the document]
Comment (katspaugh) [reply src/services/analytics/events/overview.ts:102]: I'll change it to meta 👍

File: src/components/tx-flow/flows/SafeAppsTx/index.tsx [hunk 2]
```diff
@@ -12,14 +12,20 @@ export type SafeAppsTxParams = {
   params?: SendTransactionRequestParams
 }
 
-const SafeAppsTxFlow = ({ data }: { data: SafeAppsTxParams }) => {
+const SafeAppsTxFlow = ({
```
Comment (usame-algan) [inline src/components/tx-flow/flows/SafeAppsTx/index.tsx:15]: Not sure why but I didn't see a `Transaction created` when I created one via CSV Airdrop safe app. Only `Propose Transaction` and `On-chain interaction`

**Comment by francovenica** [comment]
I compared GTM calls from this PR and dev to  see what has changed/added and I have doubts on how the table in the doc is supposed to correlate with what it was done in this ticket given the "proposed event Name"  column and "old event column"

Opening safe: There is a new event with an eventAction:"safe viewed". This seems to be added when comparing with dev. 
Now in the doc it seems there is an event for that that should be called "safe_opened" and the Old event name it was "overview", which is still the eventCategory in this PR.
So, is this something it should change? should this event should have the "safe_opened" event category?
image

Same if you open a safe after going through the Add safe flow. The eventAction is Open safe and the eventCategory is load-safe which in the table is described as "Old event name", so should this one also be changed?

**Comment by francovenica** [comment]
### Issues:
Spending_limit_transfer: I don't know how to trigger this one. In the UI we don't have a way to transfer the spending limit to another address, we can just remove and add (even the "Editing" is an "spending limit add")

Batch:
The eventLabel "batch" is shown when the batch is signed/executed, so that's fine
An issue I have is that adding a tx to the batch with the "Add to batch" button is also triggering the same event as signing the tx; so "transfer_token" if you batch a Send funds transaction or "transfer_nft" if you batch a Send NFT tx. This is something that doesn't happen in dev.
image

Safe Apps:
Tested by using the Tx builder and a dummy contract with test methods.
Not sure how to trigger this one. I assumed they should trigger when you sign a tx that was done by a safe app.
I've checked and the calls in this PR are the same as the ones in dev, so it might be that I'm just looking in the wrong place.
The events I see when creating a tx through safe apps are these 2:
1 - eventAction: "Propose Transaction" eventCategory: "safe-apps"
2 - eventAction: "Submit transactions confirmed" eventCategory "safe-apps-analytics"

### Notes:
I've noticed that we don't have an event for bulk execution.

For the WallectConnect event, I assume is when you execute a tx while connected to dApp through the WC pairing feature. I currently don't have funds so I'll try this when I can get more.

### The ones that work fine (just listed here for the sake of documentation):

Beyond what I said in the previous comment, the "safe viewed" event only triggers on opening, not anymore as you navigate through the safe, so that's the expected behavior.
The "Simulate event" looks good
The "Toggle execute tx" looks good, with a true/false eventLabel depending on what the user chooses ("yes, execute" or "no, later")

Transactions:
These trigger when you sign the tx to propose it.

Owner adding: looks good. a new event comparing dev with this PR, with an eventAction: "Create Transaction" and an eventLabel: "owner_add"
Owner removal: Looks good, same as before with eventLabel: "owner_remove"
Owner swap: Looks good. eventLabel: "owner_swap"
Change Threshold: Looks good. eventLabel: "owner_threshold_change"

Guard_remove: Looks good. eventLable: 'guard_remove'. Safe with a guard for regression later: gor:0x24F8542Da0C5f267c177a224270442C73804EE80
Module_remove: Looks good. eventLabel: 'module_remove'
Spending_limit_remove: Looks good. eventLabel: 'spending_limit_remove',
Spending_limit_add: Looks good. eventLable: 'spending_limit_add',

Safe_update: looks good. eventLable: 'safe_update',

Transfer_token. Looks good. eventLabel: 'transfer_token',
Transfer_nft: Looks good. eventLabel: 'transfer_nft',

Rejection. Looks good. eventLabel: 'rejection',
Typed_message: looks good. eventLable: 'typed_message', (tested this by connecting to OpenSea with WC, which requires a off-chain signature of a message)

**Comment by katspaugh** [comment]
This PR doesn't include the new naming scheme, this will be done later.

I'm only adding `Safe viewed` and `Transaction created` using the old naming convention.

Fixed the `batch` and `safeapps` events. ✅ 

> For the WallectConnect event, I assume is when you execute a tx while connected to dApp through the WC pairing feature. I currently don't have funds so I'll try this when I can get more.

You don't have to execute them, this new event tracks tx _creation_.

**Comment by francovenica** [comment]
The batch fixing is fine

For Safe apps I tried again and I still see the same calls I reported, they are the same as the ones I see in current Dev. So I dont see the  `safeapps = 'safeapps' event.

I use the tx builder with a dummy contract to execute a test method.
image

Maybe the current calls are the ones that the tracking expects. let me know if that's the case

**Comment by francovenica** [comment]
Same thing happens with WC.
I tried in this PR and in dev to create the same tx: using Curve to swap tokens in a polygon safe.
After creating the tx the GTM calls I had were the same in this PR and dev. So I didn't see the `walletconnect = 'walletconnect',` event
image

**Comment by katspaugh** [comment]
@francovenica thanks! Fixed for both safeapps and walletconnect. ✅

**Comment by francovenica** [comment]
Yes, the events are there now:

image

image

**Pull request merged**